Mongo connector

vurankar opened the pull request:
- ability to drop and repopulate a single index
- update Dockerfile 
   - update python 3.5
- update Jenkinsfile to build and upload mongo connector container to docker hub
- split settings and mapping config file into per index settings and mappings files

lauryl commented:
@vurankar - the commit history is a bit messy. I see a commit from Lili - perhaps upstream/6.7 needs to be rebased against the branch Lili originally committed to

vurankar commented:
Rebased 6.7 with 6.7.lili this brought in Lili's changes. 
Rebased Vishram with 6.7 . this brought in both your changes and Lili's changes. 

After merging this PR upstream 6.7 will have all consolidated search changes.
